.github/workflows/check.yml

```diff
@@ -0,0 +1,155 @@
+        # https://gitlab.haskell.org/ghc/ghc/-/issues/21111
+        # if we _do_ want them, this would be the lines to add below
+        
+        $ghcMingwDir = Join-Path -Path $(ghc --print-libdir) `
+                                 -ChildPath ../mingw/x86_64-*-mingw32/lib/ `
+                                 -Resolve
+        
+        cabal user-config -a "extra-prog-path: C:/msys64/mingw64/bin, C:/msys64/usr/bin" `
+                          -a "extra-include-dirs: C:/msys64/mingw64/include" `
+                          -a ("extra-lib-dirs: {0}, C:/msys64/mingw64/lib" -f $ghcMingwDir) `
+                          -f init
+
+    - name: "OUTPUT Record cabal-store (Linux)"
+      id: lin-setup-haskell
+      shell: bash
+      if: runner.os != 'Windows'
+      run: |
+        echo "cabal-store=/home/runner/.cabal/store" >> $GITHUB_OUTPUT
+        
+    - name: "OUTPUT Record cabal-store (Windows)"
+      id: win-setup-haskell
+      shell: bash
+      if: runner.os == 'Windows'
+      run: |
+        echo "cabal-store=C:\\cabal\\store" >> $GITHUB_OUTPUT     
+
+    - name: "[OUTPUT] cache keys: version, weeknum"
+      id: cache-keys
+      shell: bash
+      run: |
+        echo "weeknum=$(/usr/bin/date -u "+%W")" >> $GITHUB_OUTPUT
+        echo "CACHE_VERSION=20220919" >> $GITHUB_OUTPUT
+        echo "DIST_CACHE_VERSION=20221122" >> $GITHUB_OUTPUT
+
+    - name: "LINUX: Install build environment (apt-get)"
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install libsodium23 libsodium-dev
+        sudo apt-get -y install libsystemd0 libsystemd-dev
+        sudo apt-get -y remove --purge software-properties-common
+        sudo apt-get -y autoremove
+
+    - name: "LINUX: Install build environment (for secp256k1)"
+      if: runner.os == 'Linux'
+      run: sudo apt-get -y install autoconf automake libtool
+
+    - uses: actions/checkout@v3
+
+    - name: Install secp256k1
+      uses: input-output-hk/setup-secp256k1@v1
+      with:
+        git-ref: ${{ env.SECP256K1_REF }}
+        cache-version: ${{ env.SECP_CACHE_VERSION }}
+
+    - name: "Configure cabal.project.local"
+      if: runner.os != 'Windows'
+      run: |
+        cp .github/workflows/cabal.project.local.Linux cabal.project.local
+
+    - name: "Configure cabal.project.local Windows"
+      if: runner.os == 'Windows'
+      run: |
+        cp .github/workflows/cabal.project.local.Windows cabal.project.local
+
+    - name: Update Hackage and CHaP
+      run: cabal update
+
+    - name: Record dependencies
+      id: record-deps
+      run: |
+        cabal build all --dry-run
+        cat dist-newstyle/cache/plan.json | jq -r '."install-plan"[].id' | sort | uniq > dependencies.txt
+
+    - name: Cabal check
+      run: |
+        for x in $(find . -name '*.cabal' | cut -c 3-); do
+          (
+            d=$(dirname $x)
+            echo "== $d =="
+            cd $d
+            cabal check
+          )
+        done
```

Comment:
AFAIU, `cabal check` does not compile anything, so I think most of this setup in this CI workflow is not necessary.

Also, can you move this to e.g. `./scripts/ci/check-cabal.sh` such that it is easy to run locally and does not get out of sync?

Reply:
Done!